File: app/src/main/java/br/com/nicetuber/feature/home/HomeChannelAdapter.java
package br.com.nicetuber.feature.home;

import android.databinding.DataBindingUtil;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import java.util.List;

import br.com.nicetuber.R;
import br.com.nicetuber.databinding.ItemViewBinding;
import br.com.nicetuber.model.ChannelSearched;
import br.com.nicetuber.util.UIListeners;

/**
 * Created by thiagozg on 15/10/2017.
 */

public class HomeChannelAdapter extends RecyclerView.Adapter<HomeChannelAdapter.ChannelHolder> {

    private List<ChannelSearched> listChannelSearched;
    private UIListeners.OnClickListener listener;

    public HomeChannelAdapter(List<ChannelSearched> listChannelSearched, UIListeners.OnClickListener listener) {
        this.listChannelSearched = listChannelSearched;
        this.listener = listener;
    }

    @Override
    public ChannelHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        ItemViewBinding binding = DataBindingUtil.inflate(
                                    LayoutInflater.from(parent.getContext()),
                                    R.layout.item_view,
                                    parent, false);

        ChannelHolder holder = new ChannelHolder(binding);
        holder.itemView.setOnClickListener(view -> {
            int position = holder.getAdapterPosition();
            ChannelSearched channelSearched = listChannelSearched.get(position);
            listener.onClick(channelSearched);
        });

        return holder;
    }

    @Override
    public void onBindViewHolder(ChannelHolder holder, int position) {
        ChannelSearched channelSearched = listChannelSearched.get(position);
        holder.binding.setChannelSearched(channelSearched);
        holder.binding.executePendingBindings();
    }

    @Override
    public int getItemCount() {
        return listChannelSearched.size();
    }

    public static class ChannelHolder extends RecyclerView.ViewHolder {

        private ItemViewBinding binding;

        public ChannelHolder(ItemViewBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

    }

}

File: app/src/main/java/br/com/nicetuber/feature/home/HomeChannelAdapter.kt
package br.com.nicetuber.feature.home


import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import br.com.nicetuber.R
import br.com.nicetuber.model.ChannelSearched
import br.com.nicetuber.util.UIListeners
import br.com.nicetuber.util.loadCircleGlide
import kotlinx.android.synthetic.main.item_view.view.*


/**
 * Created by thiagozg on 15/10/2017.
 */

class HomeChannelAdapter(
    private val listChannelSearched: List<ChannelSearched>,
    private val listener: UIListeners.OnClickListener) : RecyclerView.Adapter<HomeChannelAdapter.ChannelHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ChannelHolder {
        val itemView = LayoutInflater.from(parent.context)
                                     .inflate( R.layout.item_view, parent, false)

        val holder = ChannelHolder(itemView)
        holder.itemView.setOnClickListener {
            val position = holder.adapterPosition
            val channelSearched = listChannelSearched[position]
            listener.onClick(channelSearched)
        }

        return holder
    }

    override fun onBindViewHolder(holder: ChannelHolder, position: Int) {
        val channelSearched = listChannelSearched[position]
        holder.bindData(channelSearched)
    }

    override fun getItemCount(): Int {
        return listChannelSearched.size
    }

    class ChannelHolder(internal val view: View) : RecyclerView.ViewHolder(view) {

        fun bindData(channelSearched: ChannelSearched) {
            itemView.iv_channel_mini.loadCircleGlide(
                    channelSearched.snippet?.thumbnails?.imageMedium?.url)
            itemView.tv_title.text = channelSearched.snippet?.title
            itemView.tv_description.text = channelSearched.snippet?.description
        }

    }

}
